include/chopper/layout/configuration.hpp

```diff
@@ -39,6 +44,30 @@ struct configuration
     bool force_all_binnings{false};
     bool output_statistics{false};
     bool debug{false};
+
+private:
+    friend class cereal::access;
+
+    template <class Archive>
+    void serialize( Archive & ar )
+    {
+      ar(CEREAL_NVP(data_file.string()));
+      ar(CEREAL_NVP(output_filename.string()));
+      ar(CEREAL_NVP(hll_dir.string()));
+      ar(CEREAL_NVP(t_max));
+    //   ar(CEREAL_NVP(aggregate_by_column));
+      ar(CEREAL_NVP(num_hash_functions));
+      ar(CEREAL_NVP(fp_rate));
+      ar(CEREAL_NVP(alpha));
+      ar(CEREAL_NVP(max_ratio));
```

Comment:
Do we need the NVP macro? What happens when we do not use it? Just wondering what the default case it and whether the default is sufficient.

Also, for `hll_dir.string()`, it should result in the var being named `hll_dir.string()` in the json. But, adding an overload for `std::filesystem::path` would solve that.

Reply:
> Do we need the NVP macro? What happens when we do not use it? Just wondering what the default case it and whether the default is sufficient.

Phew I don't know.. I tried to follow the documentation of cereal and copy and pasted an example there instead of looking how you did it in raptor.

Reply:
Update: without the NVP macro, the names are not displayed in the config file, e.g. `tmax: 64` but instead it is `value0: 64`.